### PR TITLE
[JBIDE-20000] Add validator to restrict adding readonly labels when using templates

### PR DESCRIPTION
**Pull request by jcantrill**
@adietish fixed

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/LabelKeyValidator.java [hunk 1]
```diff
@@ -24,12 +26,14 @@
 	
 	public static final int SUBDOMAIN_MAXLENGTH = 253;
 	private static final Pattern SUBDOMAIN_REGEXP = Pattern.compile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$");
+	private Collection<String> readonlykeys;
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/LabelKeyValidator.java]: shouldnt this be a set?

**Comment by adietish** [comment]
@jcantrill could we please avoid those merging commits like in this PR? I noticed this in a few PRs so far and if we can avoid them we'd have a much nicer git history?

**Comment by jcantrill** [comment]
@adietish I would love to avoid merging them but I don't know how. I thought this was a product of using Github and something it does.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/LabelKeyValidator.java (hunk 1 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/LabelKeyValidator.java]: @adietish This probably should be a set but for now there is only one value.  I dont anticipate haveing many others.

**Comment by adietish** [comment]
@jcantrill --rebase wont work? or git fetch upstream/master && git reset --hard FETCH_HEAD && ... ?
Having a "clean" history is pretty valuable imho, much easier to read and track. Worth the effort to have this sorted out I'd think.

**Comment by adietish** [comment]
ok good to go and would be great if you could rebase and remove the merge commit for improved history readability

**Comment by adietish** [comment]
rebased, merged and pushed to upstream/master

**Comment by adietish** [comment]
@jcantrill rebasing made the merge commit go away

**Pull request closed without merging**